### PR TITLE
[SPARK-49412][PS] Compute all box plot metrics in single job

### PR DESCRIPTION
**Pull request by zhengruifeng**
### What changes were proposed in this pull request?
Compute all box plot metrics in single job

### Why are the changes needed?
existing implementation computes the necessary metrics step by step:
1, compute q1/q3/mean/median;
2, compute the whiskers: min and max values of non-outliers;
3, get top 1k outliers;

so it needs 3 jobs, however these metrics can be computed in single job:
1, performance improvement: save ~20% computation time;
2, simplify the implementation, to make it easier to reuse in the future.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
CI and manually check

```
import pandas as pd
import numpy as np
from datetime import datetime

import pyspark.pandas as ps

np.random.seed(123)
pdf = pd.DataFrame(np.random.random([1000000, 5]), columns=["a", "b", "c", "d", "e"])
psdf = ps.DataFrame(pdf)
len(psdf)

for i in range(20):
    a = psdf.boxplot()

tic = datetime.now()
for i in range(10):
    a = psdf.boxplot()
datetime.now() - tic
```

before:

the avg computation duration is around 0.3 + 0.3 + 0.5 = 1.1 s,
the last 3 runs:
image

after:

the avg computation duration is around 0.8 ~ 0.9 s,
the last 3 runs:
image

### Was this patch authored or co-authored using generative AI tooling?
No

### CODE REVIEW

File: python/pyspark/pandas/plot/core.py [hunk 1]
```diff
@@ -285,209 +285,76 @@ def binary_search_for_buckets(value):
...
             )
 
-        results = multicol_outliers.select(scols).first()
-
-        fliers = {}
-        for i, colname in enumerate(colnames):
-            fliers[colname] = results[i]
-
-        return fliers
+        sdf_result = sdf.join(sdf_stats.hint("broadcast")).select(*result_scols)
```
Comment (xinrong-meng) [inline python/pyspark/pandas/plot/core.py:356]: Nice!

**Review by xinrong-meng — APPROVED** [review]

**Comment by xinrong-meng** [comment]
I think that improvement deserves to be highlighted in the release notes

**Comment by zhengruifeng** [comment]
> I think that improvement deserves to be highlighted in the release notes

thanks for reviews.
I am not sure whether we should add it in release notes since it is a refactoring and optimization.

**Review by HyukjinKwon — APPROVED** [review]

**Comment by zhengruifeng** [comment]
thanks, merged to master

**Pull request closed without merging**